commands/api/pagination.go

```diff
@@ -0,0 +1,107 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+var linkRE = regexp.MustCompile(`<([^>]+)>;\s*rel="([^"]+)"`)
+
+func findNextPage(resp *HTTPResponse) (string, bool) {
+	for _, m := range linkRE.FindAllStringSubmatch(resp.Response.Header.Get("Link"), -1) {
+		if len(m) >= 2 && m[2] == "next" {
+			return m[1], true
+		}
+	}
+	return "", false
+}
+
+func findEndCursor(r io.Reader) string {
+	dec := json.NewDecoder(r)
+
+	var idx int
+	var stack []json.Delim
+	var lastKey string
+	var contextKey string
+
+	var endCursor string
+	var hasNextPage bool
+	var foundEndCursor bool
+	var foundNextPage bool
+
+loop:
+	for {
+		t, err := dec.Token()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return ""
+		}
+
+		switch tt := t.(type) {
+		case json.Delim:
+			switch tt {
+			case '{', '[':
+				stack = append(stack, tt)
+				contextKey = lastKey
+				idx = 0
+			case '}', ']':
+				stack = stack[:len(stack)-1]
+				contextKey = ""
+				idx = 0
+			}
+		default:
+			isKey := len(stack) > 0 && stack[len(stack)-1] == '{' && idx%2 == 0
+			idx++
+
+			switch tt := t.(type) {
```

Comment:
I would suggest, we maybe flatten this switch.
The part where `isKey` is being calculated t.i. 
```go
isKey := len(stack) > 0 && stack[len(stack)-1] == '{' && idx%2 == 0
idx++
```
Then we could just remove the nested switch , and replace the default case above with the cases inside the this switch

Reply:
I just understood that I wrote this comment kinda leaving out some words. 
I intended to mean we could rewrite this as 
```go
isKey := len(stack) > 0 && stack[len(stack)-1] == '{' && idx%2 == 0
idx++
switch tt := t.(type) {
...
case string:
...
case bool:
...
}
```
This way we can remove one of the nested switches, but if you think that there is no need for this then we can just leave it as is.

Reply:
Yeah got it... Resolved now